Inc/HALALMock/Services/Communication/Ethernet/TCP/Socket.hpp

```diff
@@ -37,9 +41,8 @@ class Socket : public OrderProtocol{
 	IPV4 remote_ip;
 	uint32_t remote_port;
 	SocketState state;
-	//Change the data type of tx/rx packet_buffer
-	queue<vector<uint8_t>> tx_packet_buffer;
-	queue<vector<uint8_t>> rx_packet_buffer;
+	queue<Packet> tx_packet_buffer;
+	queue<Packet> rx_packet_buffer;
```

Comment:
this should be a reference or a pointer, otherwise you dont get polymorphism

Reply:
True